src/main/scala/services/mssql/query/LazyQueryResult.scala

```diff
@@ -17,7 +19,7 @@ import scala.util.{Failure, Success, Try}
  * @param statement The statement used to execute the query.
  * @param resultSet The result set of the query.
  */
-class LazyQueryResult(protected val statement: Statement, resultSet: ResultSet, eagerHead: List[DataRow]) extends QueryResult[LazyList[DataRow]]
+class LazyQueryResult(protected val statement: Statement, protected  val resultSet: ResultSet, eagerHead: List[DataRow]) extends QueryResult[LazyList[DataRow]]
```

Comment:
this thing is also not very effectful, esp the .next() call is sure a blocking IO one